File: matrices-multidimensionales-ll/javascript/matrices.js
const readline = require('node:readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

let personas = [];

function preguntar(pregunta) {
    return new Promise(resolve => {
        readline.question(pregunta, resolve);
    });
}

async function cargarPersona() {
    const nombre = await preguntar("Ingrese el nombre: ");
    const apellido = await preguntar("Ingrese el apellido: ");
    const dni = await preguntar("Ingrese el DNI: ");

    const telefonos = [];
    console.log("Ingrese los números de teléfono (deje en blanco para finalizar):");

    while (true) {
        const telefono = await preguntar("> ");
        if (!telefono) {
            break;
        }
        telefonos.push(telefono);
    }

    const hijos = [];
    console.log("Ingrese el nombre de sus hijos (deje en blanco para finalizar):");
    while (true) {
        const hijo = await preguntar("> ")
        if (!hijo) {
            break;
        }
        hijos.push(hijo)
    }

    return [nombre, apellido, dni, telefonos, hijos];
}

function mostrarDatos() {
    console.log("\n--- Datos de las personas cargadas ---");

    if (personas.length === 0) {
        console.log("No se han cargado datos.");
        return;
    }

    personas.forEach(persona => {
        const [nombre, apellido, dni, telefonos, hijos] = persona;

        console.log(`Nombre: ${nombre}`);
        console.log(`Apellido: ${apellido}`);
        console.log(`DNI: ${dni}`);

        if (telefonos.length > 0) {
            console.log("Teléfonos:");
            telefonos.forEach(telefono => {
                console.log(`- ${telefono}`);
            });
            return "No se registraron telefonos."
        }

        if (hijos.length > 0) {
            console.log("Hijos:");
            hijos.forEach(hijo => {
                console.log(`- ${hijo}`);
            })
            return "No se registraron nombres de hijos"
        }

        console.log("-".repeat(20));
    });
}

async function main() {
    while (true) {
        console.log("\n--- Menú ---");
        console.log("1. Cargar nueva persona");
        console.log("2. Mostrar datos cargados");
        console.log("3. Filtrar por DNI");
        console.log("4. Salir");

        const opcion = await preguntar("Seleccione una opción: ");

        if (opcion === '1') {
            const persona = await cargarPersona();
            personas.push(persona);
            console.log("Persona cargada correctamente.");
            continue;
        }

        if (opcion === '2') {
            mostrarDatos();
            continue;
        }

        if (opcion === '3') {
            const persona = await preguntar("Ingrese el DNI a filtrar: ");
            if (persona) {
                console.log("\n--- Persona encontrada ---");
                mostrarDatos([persona]);
            }
            console.log("No se encontró ninguna persona con ese DNI.")
            continue;
        }

        if (opcion === '4') {
            console.log("Saliendo");
            readline.close();
            break;
        }

        console.log("Opcion invalida. Intente nuevamente.");
    }
}


main();

File: matrices-multidimensionales-ll/python/matrices.py
def cargar_persona():
    nombre = input("Ingrese el nombre: ")
    apellido = input("Ingrese el apellido: ")
    dni = input("Ingrese el DNI: ")

    telefonos = []
    print("Ingrese los números de teléfono (deje en blanco para finalizar):")
    while True:
        telefono = input("> ")
        if not telefono:
            break
        telefonos.append(telefono)
    hijos  = []

    print('Ingrese el nombre de sus hijos (deje en blanco para finalizar):')
    while True:
        hijo = input("> ")
        if not hijo:
            break
        hijos.append(hijo)

    return [nombre, apellido, dni, telefonos, hijos]

def mostrar_datos(personas):
    print("\n--- Datos de las personas cargadas ---")

    if not personas:
        print("No se han cargado datos.")
        return

    for persona in personas:
        nombre, apellido, dni, telefonos, hijos = persona

        print(f"Nombre: {nombre}")
        print(f"Apellido: {apellido}")
        print(f"DNI: {dni}")

        if telefonos:
            print(f"Teléfonos({len(telefonos)}):")
            for telefono in telefonos:
                print(f"- {telefono}")
        else:
            print("No se registraron teléfonos.")

        if hijos:
            print(f"Hijos({len(telefonos)}):")
            for hijo in hijos:
                print(f"- {hijo}")
        else:
            print("No se registraron nombre de hijos")
        print("-" * 20)


def main():
    personas = []
    while True:
        print("\n--- Menú ---")
        print("1. Cargar nueva persona")
        print("2. Mostrar datos cargados")
        print("3. Filtrar por DNI")
        print("4. Salir")
        opcion = input("Seleccione una opción: ")

        if opcion == '1':
            persona = cargar_persona()
            personas.append(persona)
            print("Persona cargada exitosamente.")
        elif opcion == '2':
            mostrar_datos(personas)
        elif opcion == '3':
            dni = input('Ingrese DNI a filtrar: ')
            encontrado = False
            for persona in personas:
                if persona[2] == dni:
                    print("\n--- Persona encontrada ---")
                    mostrar_datos([persona])
                    encontrado = True
                    break
            if not encontrado:
                print("No se encontró ninguna persona con ese DNI.")
        elif opcion == '4':
            print("Saliendo del programa")
            break
        else:
            print("Opción inválida. Intente nuevamente.")

main()
